### PR TITLE
Add OpenRPC schema generator

### LINKED ISSUE #54: Support openrpc
https://open-rpc.org

### PR DESCRIPTION
**Pull request by qweeze**
Closes #54 

@smagafurov @spumer Please have a look. I'll add some tests if this looks good enough to you

### CODE REVIEW

File: fastapi_jsonrpc/__init__.py [hunk 1]
```diff
@@ -1318,6 +1321,77 @@ def openapi(self):
...
+                        'name': tag,
+                    }
+                    for tag in route.tags
+                ],
+                'errors': [
+                    {
+                        'code': error.CODE,
+                        'message': error.MESSAGE,
+                    }
+                    for error in route.errors
```
Comment (spumer) [inline fastapi_jsonrpc/__init__.py:1367]: What about more complicated error schema? OpenRPC allow use extra fields except `code` and `message`?

Right now fastapi-jsonrpc support custom ErrorModel which allow you extend jsonrpc error results and this custom schema is visible in openapi (Swagger)

Comment (spumer) [reply fastapi_jsonrpc/__init__.py:1367]: Yep, https://spec.open-rpc.org/#error-object allow custom data too

`BaseError` have a get_data_model() method. You can use it to pass correct schema into openRPC data part

Comment (qweeze) [reply fastapi_jsonrpc/__init__.py]: Thanks, custom errors' schemas are now supported https://github.com/smagafurov/fastapi-jsonrpc/pull/56/commits/818c1582fa5123f50298fca0157570027b96ed00

**Comment by spumer** [comment]
LGTM, other minor improvements like cache and route override/customize schema we can do later. 

You can start make tests 🔥

**Comment by qweeze** [comment]
All right, now we have some tests f9cb6cb65b32a9c4afad947e51a72b2345cf06c5
Let me know if you'd like to add any other test cases

**Comment by spumer** [comment]
> All right, now we have some tests [f9cb6cb](https://github.com/smagafurov/fastapi-jsonrpc/commit/f9cb6cb65b32a9c4afad947e51a72b2345cf06c5) Let me know if you'd like to add any other test cases

Have a openRPC examples support?

I mean if we have two error with the same code like `(6001, ValidationError)` but have few different examples with different `data` field. It's possible include this into openRPC?

**Comment by qweeze** [comment]
> Is there support for examples in openRPC?

There is, but it's meant to be used only for params and result values. As for errors, the only place to store them is `errors` array inside `method` objects, and they're required to be unique. Spec explicitly states that `The Errors MUST have unique error codes`. 

At the same time, `error.data` field is not typed, so technically it's OK to "merge" such errors without abusing the spec like this:
```json
{
    "error": 6000,
    "message": "ValidationError",
    "data": {
         "anyOf": [
            {"title": "MyError.Data", "type": "object", "properties": {"details": {"title": "Details", "type": "string"}}, "required": ["details"]},
            {"title": "MyError.Data", "type": "string"}
          ]
    }
}
```

**Comment by spumer** [comment]
Can you make it? But i think current implementation of openRPC playground does not supported it.

Other things looks good for me

**Comment by qweeze** [comment]
> Can you make it? 

I had to move errors' defs into `components.errors`, this way tools can resolve `$ref`s in error body. Anyway, `data` schemes of same-coded errors are now joined with `anyOf`, I added a test for this case [here](https://github.com/qweeze/fastapi-jsonrpc/blob/master/tests/test_openrpc.py#L241)

Seems that playground has no problem with rendering this as well:
image

**Pull request merged**